### ISSUE TITLE
minikube mount failing on Mac OS and VirtualBox

### ISSUE THREAD

**Issue by fahanson** [kubernetes/minikube#2947]
**Bug Report**

Please provide the following details:

**Environment**:

**Minikube version** v0.28.0
- **OS** Mac OS 10.13.5
- **VM Driver**: VirtualBox
- **ISO version** :
- **Install tools**:
- **Others**: VirtualBox version 5.2.12 r122591 (Qt5.6.3)

**What happened**:
Running `minikube mount` fails silently. When expanding the verbosity I am seeing ssh fail:
```
Using SSH client type: native
&{{{<nil> 0 [] [] []} docker [0x143a220] 0x143a1d0  [] 0s} 127.0.0.1 50452 <nil> <nil>}
About to run SSH command:
sudo umount /Users/faddai;
SSH cmd err, output: <nil>:
Using SSH client type: native
&{{{<nil> 0 [] [] []} docker [0x143a220] 0x143a1d0  [] 0s} 127.0.0.1 50452 <nil> <nil>}
About to run SSH command:
sudo mkdir -p /Users/faddai || true;
sudo mount -t 9p -o trans=tcp,port=51934,dfltuid=1001,dfltgid=1001,version=9p2000.u,msize=262144 192.168.99.1 /Users/faddai;
sudo chmod 775 /Users/faddai;
2018/06/28 20:38:44 connected
2018/06/28 20:38:44 >>> 192.168.99.100:35446 Tversion tag 65535 msize 65536 version '9P2000.u'```
2018/06/28 20:38:44 <<< 192.168.99.100:35446 Rversion tag 65535 msize 65536 version '9P2000.u'
2018/06/28 20:38:44 >>> 192.168.99.100:35446 Tattach tag 1 fid 0 afid 4294967295 uname 'nobody' nuname 4294967295 aname ''
2018/06/28 20:38:44 <<< 192.168.99.100:35446 Rattach tag 1 aqid (1750fb 481e9a02 'd')
2018/06/28 20:38:44 >>> 192.168.99.100:35446 Tstat tag 1 fid 0
2018/06/28 20:38:44 <<< 192.168.99.100:35446 Rstat tag 1 st ('faddai' 'none' 'none' 'none' q (1750fb 481e9a02 'd') m d775 at 0 mt 1530218322 l 1568 t 0 d 0 ext )
2018/06/28 20:38:44 >>> 192.168.99.100:35446 Tattach tag 1 fid 1 afid 4294967295 uname '' nuname 0 aname ''
2018/06/28 20:38:44 <<< 192.168.99.100:35446 Rattach tag 1 aqid (1750fb 481e9a02 'd')
2018/06/28 20:38:44 >>> 192.168.99.100:35446 Tstat tag 1 fid 1
2018/06/28 20:38:44 <<< 192.168.99.100:35446 Rstat tag 1 st ('faddai' 'none' 'none' 'none' q (1750fb 481e9a02 'd') m d775 at 0 mt 1530218322 l 1568 t 0 d 0 ext )
2018/06/28 20:38:44 >>> 192.168.99.100:35446 Twstat tag 1 fid 1 st ('' '' '' '' q (ffffffffffffffff ffffffff 'daAltL') m d775 at 4294967295 mt 4294967295 l 18446744073709551615 t 65535 d 4294967295 ext )
2018/06/28 20:38:44 <<< 192.168.99.100:35446 Rwstat tag 1
SSH cmd err, output: <nil>:
```

**What you expected to happen**:
Expect minikube to successfully mount the directory into the container.

**How to reproduce it** (as minimally and precisely as possible):
`minikube mount -v 15 /Users/addai:/Users/addai`

**Anything else do we need to know**:
No firewall on this machine. VirtualBox IP settings on the hostonly adapter all seem correct.

**Comment by kgignatyev-inspur** [comment]
/Users is not mounted on host, so nothing is visible to containers

**Comment by dlorenc** [comment]
Fixed in #2986
